src/osd/modules/render/drawbgfx.cpp

```diff
@@ -382,6 +386,28 @@ bool video_bgfx::init_bgfx_library(osd_window &window)
 }
 
 
+//============================================================
+//  Helper for creating a wayland window
+//============================================================
+
+#if defined(SDLMAME_USE_WAYLAND)
+wl_egl_window* create_wl_egl_window(SDL_Window *window, struct wl_surface *surface)
+{
+	wl_egl_window *win_impl = (wl_egl_window*)SDL_GetWindowData(window, "wl_egl_window");
+	if(!win_impl)
+	{
+		int width, height;
+		SDL_GetWindowSize(window, &width, &height);
+		if(!surface)
+			return nullptr;
+		win_impl = wl_egl_window_create(surface, width, height);
+		SDL_SetWindowData(window, "wl_egl_window", win_impl);
+	}
+	return win_impl;
+}
+#endif
```

Comment:
Please follow the formatting of the file:
* Space before but not after right-associative unary `*`
* Space between flow control keyword and opening parenthesis of controlling expression

Why are you checking `surface` *after* getting the window size?  You won’t use the window size if it’s null.

Can `wl_egl_window_create` fail?

Unlike the other cases in the `switch (wmi.subsystem)` this can fail.  However, you aren’t checking for failure and returning `false` from `video_bgfx::set_platform_data` when that happens.

Reply:
Should be fixed now.